### ISSUE TITLE
Invalid emit for 'export default function' in namespace

### ISSUE THREAD

**Issue by ajafff** [Microsoft/TypeScript#16302]
**TypeScript Version:**  2.3.4 and current playground

**Code**

```ts
namespace ns {
    export default function foo() {} // no error, wrong emit
}

```

**Expected behavior:**

Error at declaration that default exports are only allowed in ECMAScript modules.

**Actual behavior:**

No error at declaration and invalid emit:
```js
var ns;
(function (ns) {
    default function foo() { } // no error, wrong emit
    ns.foo = foo;
})(ns || (ns = {}));
```

**Comment by mhegazy** [comment]
This is an error, and should be flagged as one. https://github.com/Microsoft/TypeScript/issues/16301 tracks that. there are no guarantees on the correctness of the output with invalid input.
